File: easy/mime-type/mime-type.py
import sys
import math

n = int(input())  # Number of elements which make up the association table.
q = int(input())  # Number Q of file names to be analyzed.

mime_type = {None : "UNKNOWN"}
for i in range(n):
    # ext: file extension
    # mt: MIME type.
    ext, mt = input().split()
    if ext.lower() not in mime_type.keys():
        mime_type[ext.lower()] = mt

for i in range(q):
    filename = input()  # One file name per line.
    print("filename:" + filename, file=sys.stderr)

    index = filename.rfind(".")
    ext = None if index == -1 else filename[index + 1:].lower()
    print(mime_type.get(ext, "UNKNOWN"))

File: easy/mime-type/mime-type.cpp
#include <iostream>
#include <string>
#include <algorithm>
#include <map>

using namespace std;

void toLower(string& str)
{
    transform(str.begin(), str.end(), str.begin(), ::tolower);
}

int main()
{
    int nbExtension; // Number of elements which make up the association table.
    cin >> nbExtension; cin.ignore();
    int nbFile; // Number of file names to be analyzed.
    cin >> nbFile; cin.ignore();

    map<string, string> mimeType;
    for (int i = 0; i < nbExtension; i++) {
        string ext; // file extension
        string mt; // MIME type.
        cin >> ext >> mt; cin.ignore();
        toLower(ext);
        mimeType.insert(pair<string, string>(ext, mt));
        cerr << "ext:" << ext << ", mt" << mt << endl;
    }

    for (int i = 0; i < nbFile; i++) {
        string filename; // One file name per line.
        getline(cin, filename);
        cerr << "filename:" << filename << endl;
        size_t index = filename.rfind('.');
        map<string, string>::iterator it =  mimeType.end();
        if (index != string::npos) {
            string ext = filename.substr(index + 1);
            toLower(ext);
            it = mimeType.find(ext);
        }
        string type = (it != mimeType.end()) ? it->second : "UNKNOWN";
        cout << type << endl;
    }
}
